core/src/main/scala/org/apache/spark/deploy/security/ServiceCredentialProvider.scala

```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.spark.deploy.yarn.security
+package org.apache.spark.deploy.security
 
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
```

Comment:
For backwards compat you'd have to call `ServiceLoader` twice, to load implementations of the two interfaces. That should be most of it, really, assuming the interface itself is not changing.